macropad/code/keys.py

```diff
@@ -0,0 +1,197 @@
+=============
+
+It can act as a single key, for example:
+
+>   ("Shoot", Keycode.ENTER),
+
+This will create a key called "Shoot" that acts exactly like
+the Enter key on a keyboard. You can hold the button down and
+get repeated presses, and when you release the button on the
+macropad, it'll stop pressing the Enter key.
+
+Take a look at
+https://docs.circuitpython.org/projects/hid/en/latest/api.html#adafruit-hid-keycode-keycode
+for the complete list of keys. 
+
+2. Several Keys
+===============
+
+It can act as several keys at once. For example:
+
+> ("Undo", (Keycode.CONTROL, Keycode.Z)),
+
+Note that the list of keys are additionally surrounded by
+brackets. That's really important.
+
+This uses the same key codes as "1. Single Key", but will
+press both those keys and won't release them until you stop
+pressing the button
+
+3. Type some text
+=================
+
+It can type a load of text for you. For example:
+
+>   ("Sign", "\n\nThanks,\nJoel"),
+
+When pressed, this will type those characters. The \n means
+a new line (i.e. press enter)
+
+This assumes your machine is setup with a normal windows
+keyboard. If you're using something else you might get
+funny behaviour.
+
+4. Sequence of keys
+=================
+
+This one's a bit more compliated to set up, but also more
+powerful! It can run a function with a series of the three
+previous options, so it can send a series of button presses
+sequentially. For example:
+
+>   def launch_command_line(kbd, layout):
+>       kbd.send(Keycode.WINDOWS, keycode.R)
+>       time.sleep(0.1)
+>       layout.write("cmd")
+>       kbd.send(Keycode.ENTER)
+
+This will create the function to run. It takes two arguments,
+the kbd argument used for individual key presses, and the
+layout argument used for sending text.
+
+The function then presses and releases the WINDOWS and R keys,
+waits for 0.1 seconds, types "cmd", and then presses ENTER.
+
+To assign that function to a key, You'd put this:
+
+>   ("cmd", launch_command_line),
+
+in the page.
+
+5. Do nothing
+=============
+
+Sometimes it's helpful to have a key do nothing when pressed.
+That would look like this:
+
+>   ("", None),
+
+You can still give it a name if you want.
+
+
+Limitations
+===========
+
+There are some limitations:
+
+ * Key names can only be up to 5 characters in length
+ 
+ * You can have up to 16 pages (beyond 4 pages the LEDs on
+   the board will switch into binary mode)
+   
+ * You can only have up to 8 buttons per page. If you
+   specify fewer, the keys will default to doing nothing.
+   
+
+'''
+
+# Don't change these lines!
+from adafruit_hid.keycode import Keycode
+from adafruit_hid.consumer_control_code import ConsumerControlCode
+import time
+
+"""
+This allows you to change the function of the push button in the
+rotary encoder. Uncomment (delete the # in front of) the one you
+want to happen.
+"""
+#rotary_press_function = ConsumerControlCode.MUTE
+#rotary_press_function = ConsumerControlCode.PLAY_PAUSE
```

Comment:
If neither is uncommented does the push button do nothing like the macro keys default to?

Reply:
No, it has a default value (PLAY_PAUSE) but this allows you to override/specify it.